### PR TITLE
api: add a metric policy filter mechanism

### PR DESCRIPTION
**Pull request by jd**
Use the auth_helper plugin mechanism to filter metric listing rather than an
hardcoded rule.

The Keystone rule is identical than before and filters based on
created_by_project_id fake-attribute.

The basic/remoteuser rule is changed to mimic the one for resources, i.e., no
filtering is enforced.

### CODE REVIEW

File: gnocchi/tests/functional/gabbits/metric-list.yaml [hunk 1]
```diff
@@ -89,7 +89,7 @@ tests:
     - name: list metrics
       GET: /v1/metric
       response_json_paths:
-          $.`len`: 2
+          $.`len`: 4
```
Comment (chungg) [inline gnocchi/tests/functional/gabbits/metric-list.yaml]: tbh, i don't really understand what the goal is. i don't think this makes sense that user foobar can list metrics from user foobaz and user jd.

we should probably add more tests to show what you're trying to achieve.

Comment (jd) [reply gnocchi/tests/functional/gabbits/metric-list.yaml]: I hoped the commit message was pretty explicit.

Comment (chungg) [reply gnocchi/tests/functional/gabbits/metric-list.yaml]: uh... so do we have the equivalent keystone tests to ensure that path still does what it should?

Comment (jd) [reply gnocchi/tests/functional/gabbits/metric-list.yaml]: Yes, it's in `test_rest`, that's the only place we test ACL.

**Pull request merged**